### DISCUSSION TITLE
Issues after upgrade to 2.3.60

### DISCUSSION THREAD

**Discussion by facyber** [Security-Onion-Solutions/securityonion#4754] [Unsupported Versions] [answered]
Hello everyone,

For the moment we have two same environments, Manger + 2 Heavy Nodes (All CentOS). Today I upgraded them both to 2.3.60 and one is working fine, but other one is Generating an error "Request failed with status code 500" when I try to open SOC.

First I waited for half an hour, thought maybe it needed time to apply updates to other nodes, but then issue was still there. So I tried to reboot servers and to apply run `sudo salt-call state.highstate` but no help.

In the `sensoroni-server.log` I see some messages like these but can not find what's that about:

```
timestamp=2021-07-09T09:38:09.361113214Z level=warn message="Host not found in raid status metrics" host=manager01
timestamp=2021-07-09T09:38:09.361206571Z level=warn message="Host not found in raid status metrics" host=heavynode01
timestamp=2021-07-09T09:38:10.176034162Z level=warn message="Host not found in raid status metrics" host=heavynode02
```

`so-status` shows all green (except on one node where I disabled steno).

Weird also is that version in SOC is shown as 0.0.0 and I don't see anything in the grid.

![image](https://user-images.githubusercontent.com/8742227/125058692-d22a9400-e0aa-11eb-8452-626fccc49a4d.png)

If there are more things I can provide/check, do tell me.

**Comment by facyber** [comment]
In addition, something from ES logs:

```
[2021-07-09T09:30:02,224][WARN ][rest.suppressed          ] path: /.reporting-*/_search, params: {index=.reporting-*}
org.elasticsearch.cluster.block.ClusterBlockException: blocked by: [SERVICE_UNAVAILABLE/1/state not recovered / initialized];
[2021-07-09T09:30:02,933][WARN ][org.elasticsearch.cluster.service.ClusterApplierService] failed to notify ClusterStateListener
java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.String (java.lang.Integer and java.lang.String are in module java.base of loader 'bootstrap')
[2021-07-09T09:30:04,322][DEPRECATION][org.elasticsearch.deprecation.rest.RestController] Legacy index templates are deprecated in favor of composable templates.
[2021-07-09T09:30:04,644][ERROR][org.elasticsearch.xpack.transform.transforms.TransformPersistentTasksExecutor] Failed to gather field mappings for index [metrics-endpoint.metadata_current_default]
org.elasticsearch.action.NoShardAvailableActionException: No shard available for [org.elasticsearch.action.fieldcaps.FieldCapabilitiesIndexRequest/unset]
```

**Comment by jertel** [comment]
The sensoroni-server.log lines would not cause a 500 error in SOC, and the ES error also wouldn't cause that problem in SOC. There will either be errors (not warns) or the errors will be in nginx. You can also look at it from the browser side, using dev tools -> Network tab, to review the requests that are failing. Also check `so-status` on the manager nodes.

**Reply by facyber** [reply]
Hi @jertel 

so-status shows all good on all 3 VMs.

Lookin in dev tools, I see error `XHR GET https://so-domain.com/api/info`, request is empty and if I open this page I get error `Unable to determine logged in user from context`. I tried also creating a new user, but issue is still there.

I have second environment who except IP and DNS, should have the exact same configuration, and there is no issue with that one.

Edit: Nginx logs does not show any error or specific message except `[13/Jul/2021:08:01:44 +0000] "POST /sensoroniagents/api/node HTTP/1.1" 200 20 "-" "Go-http-client/1.1" "-"`

**Reply by jertel** [reply]
Are these two environments (the one that works and the one that doesn't) on different networks or subnets? It seems that something in your network is stripping out headers from the SOC browser requests.

**Reply by facyber** [reply]
They are in the same subnet (different vlans of course) but hosted on different servers. There were no changes in the networks during the upgrade. And it is only SOC affected as it seems. When I open kibana (from history) I can view all data and I can see logs are coming, that confuses me even more.

**Reply by jertel** [reply]
In a new incognito browser tab, open DevTools and view the Network tab. Now perform a login to SOC. Check those network requests / responses. Most likely some requests are failing to reach the server.

**Reply by facyber** [reply]
The only not okay status I see is for `/api/info` page as mentioned before, for that page I am getting status code 500 for GET, and if I try to open it I get that `Unable to determine logged in user from context` message.

I compared same GET messages for these two environments and the only difference is in Response headers, where I actually see the version is fine, but content-type and content-length are different.

```
HTTP/2 500 Internal Server Error
server: nginx
date: Thu, 15 Jul 2021 09:42:59 GMT
content-type: text/plain; charset=utf-8
content-length: 47
version: 2.3.60
X-Firefox-Spdy: h2
```

```
HTTP/2 200 OK
server: nginx
date: Thu, 15 Jul 2021 09:46:48 GMT
content-type: application/json
content-encoding: gzip
version: 2.3.60
X-Firefox-Spdy: h2
```

Also further comparison from network tab, is that after error 500 I only see two additional GET requests, while on the working one after same two requests goes request for `/api/users/`. Opening this page works fine, but for some reason `api/info` can not be opened where I presume are crucial information for authentication process.

Thanks

**Comment by facyber** [comment, marked as answer]
Hi @jertel 

It seems there was issue with nginx configuration. As we copied to `/opt/so/saltstack/local/salt/nginx/etc/` in order to make changes, they were quite different as it seems from original version and the one from 2.3.60. I copied again the new nginx.config and edited it with our need and it works now.

Now it makes sense why it does work with other environment, the other one was deployed from 2.3.50 version, while the one that not worked was from 2.3.30 if I remember correctly.

It seems I will need to review all files now that I changed/copied so far, and I do know that there is a plan to make this better in the future, and for now manual work is needed.

Thanks for the support and help.

**Reply by jertel** [reply]
Thanks for keeping us updated on this. In our next release we're planning to show a reminder to the user after soup completes, if soup detects any customizations that might need manually merged.
